cpp/tests/io/json/json_test.cpp

```diff
@@ -2159,6 +2159,84 @@ TEST_F(JsonReaderTest, JSONLinesRecoveringSync)
   cudf::set_pinned_memory_resource(last_mr);
 }
 
+// Validation
+TEST_F(JsonReaderTest, ValueValidation)
+{
+  // parsing error as null rows
+  std::string data =
+    // 0 -> a: -2 (valid)
+    R"({"a":-2 }{})"
+    "\n"
+    // 1 -> (invalid)
+    R"({"b":{}should_be_invalid})"
+    "\n"
+    // 2 -> b (valid)
+    R"({"b":{"a":3} })"
+    "\n"
+    // 3 -> c: (valid/null based on option)
+    R"({"a": 1, "c":nan, "d": "null" } )"
+    "\n"
+    "\n"
+    // 4 -> (valid/null based on option)
+    R"({"a":04, "c": 1.23, "d": "abc"} 123)"
+    "\n"
+    // 5 -> (valid)
+    R"({"a":5}//Comment after record)"
+    "\n"
+    // 6 -> ((valid/null based on option)
+    R"({"a":06} //Comment after whitespace)"
+    "\n"
+    // 7 -> (invalid)
+    R"({"a":5 //Invalid Comment within record})";
+
+  // leadingZeros allowed
+  // na_values,
+  {
+    cudf::io::json_reader_options in_options =
+      cudf::io::json_reader_options::builder(cudf::io::source_info{data.data(), data.size()})
+        .lines(true)
+        .recovery_mode(cudf::io::json_recovery_mode_t::RECOVER_WITH_NULL);
+    cudf::io::table_with_metadata result = cudf::io::read_json(in_options);
+
+    EXPECT_EQ(result.tbl->num_columns(), 4);
+    EXPECT_EQ(result.tbl->num_rows(), 8);
+    auto b_a_col = int64_wrapper({0, 0, 3, 0, 0, 0, 0, 0});
+    auto a_column =
+      int64_wrapper{{-2, 0, 0, 1, 4, 5, 6, 0}, {true, false, false, true, true, true, true, false}};
```

Comment:
@revans2 index a[4] should null right?
Since the validation failed, the entire row should be null.

Reply:
Yes, but you have not enabled the stricter validation.

```
      cudf::io::json_reader_options::builder(cudf::io::source_info{data.data(), data.size()})
        .lines(true)
        .recovery_mode(cudf::io::json_recovery_mode_t::RECOVER_WITH_NULL)
        .strict_validation(true);
```

If you don't set strict_validation it falls back to not checking.

Reply:
```
scala> Seq("""{"a":-2 }{}""", """{"b":{}should_be_invalid}""","""{"b":{"a":3} }""","""{"a": 1, "c":nan, "d": "null" }""","""{"a":04, "c": 1.23, "d": "abc"} 123""","""{"a":5}//Comment after record""","""{"a":06} //Comment after whitespace""","""{"a":5 //Invalid Comment within record}""").toDF("json").selectExpr("*","get_json_object(json,'$') as checked").show(false)
+---------------------------------------+-------------+
|json                                   |checked      |
+---------------------------------------+-------------+
|{"a":-2 }{}                            |{"a":-2}     |
|{"b":{}should_be_invalid}              |null         |
|{"b":{"a":3} }                         |{"b":{"a":3}}|
|{"a": 1, "c":nan, "d": "null" }        |null         |
|{"a":04, "c": 1.23, "d": "abc"} 123    |null         |
|{"a":5}//Comment after record          |{"a":5}      |
|{"a":06} //Comment after whitespace    |null         |
|{"a":5 //Invalid Comment within record}|null         |
+---------------------------------------+-------------+
```

With the above query a null output means it is invalid, but that only takes into account the setting for `get_json_object` in Spark.

Reply:
`nan` is not a valid literal, only `NaN` is valid. So, that line validation fails. so, it should be null, not `1`.
Do that sound right?

Reply:
That's right. The valid non-numeric numbers are (case sensitive): `"NaN", "+INF", "-INF", "+Infinity", "Infinity", "-Infinity"`.